src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Arrays.cs

```diff
@@ -325,6 +326,264 @@ public async Task MapAction_ImplicitQuery_NullableStringArrayParam()
+    }
+
+    [Fact]
+    public async Task MapAction_ImplicitQuery_IntArrayParam_Optional()
+    {
+        var (results, compilation) = await RunGeneratorAsync("""
+            app.MapGet("/hello", (int[]? p = null) => p);
+            """);
+
+        var endpoint = GetEndpointFromCompilation(compilation);
+
+        VerifyStaticEndpointModel(results, endpointModel =>
+        {
+            Assert.Equal("MapGet", endpointModel.HttpMethod);
+        });
+
+        var httpContext = CreateHttpContext();
+        httpContext.Request.QueryString = new QueryString("?p=0&p=1&p=2");
+
+        await endpoint.RequestDelegate(httpContext);
+        await VerifyResponseBodyAsync(httpContext, "[0,1,2]");
+        await VerifyAgainstBaselineUsingFile(compilation);
+    }
+
+    [Fact]
+    public async Task MapAction_ImplicitQuery_IntArrayParam_Optional_QueryNotPresent()
+    {
+        var (results, compilation) = await RunGeneratorAsync("""
+            app.MapGet("/hello", (int[]? p = null) => p);
+            """);
+
+        var endpoint = GetEndpointFromCompilation(compilation);
+
+        VerifyStaticEndpointModel(results, endpointModel =>
+        {
+            Assert.Equal("MapGet", endpointModel.HttpMethod);
+        });
+
+        var httpContext = CreateHttpContext();
+
+        await endpoint.RequestDelegate(httpContext);
+        await VerifyResponseBodyAsync(httpContext, "[]");
+        await VerifyAgainstBaselineUsingFile(compilation);
+    }
+
+    [Fact]
+    public async Task MapAction_ExplicitQuery_IntArrayParam_Optional()
+    {
+        var (results, compilation) = await RunGeneratorAsync("""
+            app.MapGet("/hello", ([FromQuery] int[]? p = null) => p);
+            """);
+
+        var endpoint = GetEndpointFromCompilation(compilation);
+
+        VerifyStaticEndpointModel(results, endpointModel =>
+        {
+            Assert.Equal("MapGet", endpointModel.HttpMethod);
+        });
+
+        var httpContext = CreateHttpContext();
+        httpContext.Request.QueryString = new QueryString("?p=0&p=1&p=2");
+
+        await endpoint.RequestDelegate(httpContext);
+        await VerifyResponseBodyAsync(httpContext, "[0,1,2]");
+        await VerifyAgainstBaselineUsingFile(compilation);
+    }
+
+    [Fact]
+    public async Task MapAction_ExplicitQuery_IntArrayParam_Optional_QueryNotPresent()
+    {
+        var (results, compilation) = await RunGeneratorAsync("""
+            app.MapGet("/hello", ([FromQuery] int[]? p = null) => p);
+            """);
+
+        var endpoint = GetEndpointFromCompilation(compilation);
+
+        VerifyStaticEndpointModel(results, endpointModel =>
+        {
+            Assert.Equal("MapGet", endpointModel.HttpMethod);
+        });
+
+        var httpContext = CreateHttpContext();
+
+        await endpoint.RequestDelegate(httpContext);
+        await VerifyResponseBodyAsync(httpContext, "[]");
+        await VerifyAgainstBaselineUsingFile(compilation);
```

Comment:
We can probably exclude the baseline here since the generated code between `MapAction_ExplicitQuery_IntArrayParam_Optional_QueryNotPresent` and `MapAction_ExplicitQuery_IntArrayParam_Optional` is the same.

Alternatively, we could do one test method that uses a `[Theory]` to test the scenario where a query is present and the one where it is not.